solc/CommandLineInterface.cpp

```diff
@@ -584,6 +587,37 @@ bool CommandLineInterface::parseLibraryOption(string const& _input)
 	return true;
 }
 
+//read file contents from m_sourceCodes into m_sourceJsons, which either are
+// a JSON-file as produced by
+// a) --combined-json -ast,compact-format
+// b) the standard-json output
+map<string, Json::Value const*> CommandLineInterface::parseAstFromInput()
+{
+	map<string, Json::Value const*> sourceJsons;
+	map<string, string> tmp_sources; //used to generate the onchainmetadata-hash
+	for (auto const& srcPair: m_sourceCodes)
+	{
+		Json::Value* ast = new Json::Value();
+		// try parsing as json
+		if (jsonParse(srcPair.second, *ast))
+			if (ast->isMember("sources"))
+				for (auto& src: (*ast)["sources"].getMemberNames())
+				{
+					std::string astKey = (*ast)[src].isMember("ast") ? "ast" : "AST";
+					astAssert( (*ast)["sources"][src][astKey]["nodeType"].asString() == "SourceUnit",  "Top-level node should be a 'SourceUnit'");
+					sourceJsons[src] = &((*ast)["sources"][src][astKey]);
+					tmp_sources[src] = dev::jsonCompactPrint(*ast);
+				}
+			else
+				BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("Invalid Format for import-JSON: Must have 'sources'-object "));
+		else
+			BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("Input file could not be parsed to JSON"));
+	}
+	//replace the internal map so that every source has its own entry
```

Comment:
Is this really still needed?

Reply:
hm. no i guess not. I guess if ```m_sourcesCodes```  is used later on to create the metadatahash, it can as well be the original string from which the JSONs were parsed. Actually that makes more sense than using an intermediate product such as the parsed and prettyPrintedJson...